src/nifty.py

```diff
@@ -1406,10 +1406,10 @@ def process_err(read):
                     read = ''.encode('utf-8')
                     while True:
                         if read_nbytes == 0:
-                            read += fh.read(rbytes)
+                            read += fh.read1(rbytes)
                             read_nbytes += rbytes
                         else:
-                            read += fh.read(1)
+                            read += fh.read1(1)
```

Comment:
Very interesting. :) What does this do?

Reply:
It solves the OSX hanging problem by having it only attempt to read the stream once. My guess is that, if `read` is used instead of `read1`, it will repeatedly attempt the read until the desired number of bytes (default here is 1024) come out. So, there's either a difference between the `read` function on mac vs linux, or `Tinker` may have slightly different behavior.

Either way, reducing `rbytes` from `1024` to some value closer to `500` fixes it on mac (probably because the total output size is less than 1024 bytes in some cases), but I wasn't very satisfied with that solution. After some more digging, I found [`read1`](https://docs.python.org/3/library/io.html#io.BufferedIOBase.read1), which attempts to get the desired number of bytes using a single read operation, and just returns what it was able to get otherwise. 

I'm interested in getting the test matrix running again so we can assess if this sneakily broke something else in the process.